src/Microsoft.ML.Recommender/MatrixFactorizationPredictor.cs

```diff
@@ -422,14 +422,18 @@ DataViewRow ISchemaBoundRowMapper.GetRow(DataViewRow input, IEnumerable<DataView
     /// <summary>
     /// Trains a <see cref="MatrixFactorizationModelParameters"/>. It factorizes the training matrix into the product of two low-rank matrices.
     /// </summary>
-    public sealed class MatrixFactorizationPredictionTransformer : PredictionTransformerBase<MatrixFactorizationModelParameters>
+    public sealed class MatrixFactorizationPredictionTransformer : PredictionTransformerBase<MatrixFactorizationModelParameters>, ISingleFeaturePredictionTransformer<MatrixFactorizationModelParameters>
     {
         internal const string LoaderSignature = "MaFactPredXf";
         internal string MatrixColumnIndexColumnName { get; }
         internal string MatrixRowIndexColumnName { get; }
         internal DataViewType MatrixColumnIndexColumnType { get; }
         internal DataViewType MatrixRowIndexColumnType { get; }
 
+        public string FeatureColumnName => throw new NotImplementedException();
```

Comment:
Are you really a `ISingleFeaturePredictionTransformer` if you just throw for all the members of `ISingleFeaturePredictionTransformer` ?

Reply:
Good question!! That brings to another topic ( In `ITrainerEsitmator<ITransformer, IModel>` Does `ITransformer` has to be a `ISingleFeaturePredictionTransformer`?). I feel like the answer is no here, and `MatrixFactorizationPredictionTransformer` shouldn't implement that `ISingleFeaturePredictionTransformer` interface. So although it seems to implement `ISingleFeaturePredictionTransformer`, it actually not. And we need to change that

The reason why currently it implement that interface is because now `AutoML` requires all it's trainers to be `ITrainerEsitmator<ITransformer, IModel>`. So in order to add `MatrixFactorization` to AutoML, it has to be a `ISingleFeaturePredictionTransformer`

Reply:
> AutoML requires all it's trainers to be ITrainerEsitmator<ITransformer, IModel>

Maybe AutoML needs to change then.

@Dmitry-A @srsaggam @vinodshanbhag @justinormont

Reply:
@daholste (once he's back) may also know why it is a `ITrainerEstimator<ITransformer, IModel>`; and the impact of changing it.